src/main/webapp/app/shared/util/core-drive-annotation-submission/core-drive-annotation-submission.ts

```diff
@@ -0,0 +1,136 @@
+      if (shouldExclude(onlyReviewedContent, tumor.cancerTypes_review)) {
+        tempTumors.push(tumor);
+        continue;
+      }
+      // process tumor cancerTypes
+      processData(tumor, ['summary', 'diagnosticSummary', 'prognosticSummary'], excludeComments, onlyReviewedContent);
+      processData(tumor.diagnostic, ['level', 'description', 'relevantCancerTypes'], excludeComments, onlyReviewedContent);
+      processData(tumor.prognostic, ['level', 'description', 'relevantCancerTypes'], excludeComments, onlyReviewedContent);
+      for (const ti of tumor.TIs) {
+        processData(ti, ['description'], excludeComments, onlyReviewedContent);
+        type TempTreatment = Omit<Treatment, 'name'> & { name: ReturnType<typeof drugUuidToDrug> | string };
+        const tempTreatments: TempTreatment[] = [];
+        for (const treatment of ti.treatments ?? []) {
+          if (shouldExclude(onlyReviewedContent, treatment.name_review)) {
+            tempTreatments.push(treatment);
+            return true;
+          }
+          (treatment as TempTreatment).name = drugUuidToDrug(treatment.name, drugList);
+          processData(
+            treatment,
+            ['level', 'propagation', 'propagationLiquid', 'indication', 'description'],
+            excludeComments,
+            onlyReviewedContent,
+          );
+        }
+        for (const item of tempTreatments) {
+          const index = ti.treatments.indexOf(item as Treatment);
+          if (index !== -1) {
+            ti.treatments.splice(index, 1);
+          }
+        }
+      }
+    }
+    for (const tumor of tempTumors) {
+      const index = mutation.tumors.indexOf(tumor);
+      if (index !== -1) {
+        mutation.tumors.splice(index, 1);
+      }
+    }
+  }
+  for (const mutation of tempMutations) {
+    const index = gene.mutations.indexOf(mutation);
+    if (index !== -1) {
+      gene.mutations.splice(index, 1);
+    }
+  }
+  return gene;
+}
+
+function processData(data: object | undefined, keys: string[], excludeComments: boolean, onlyReviewedContent: boolean) {
+  if (data !== undefined) {
+    for (const key of keys) {
+      if (excludeComments) {
+        delete data[key + '_comments'];
+      }
+      const reviewKey = key + '_review';
+      const maybeReview = data[reviewKey];
+      if (onlyReviewedContent && typeof maybeReview === 'object' && maybeReview !== null && 'lastReviewed' in maybeReview) {
+        data[key] = maybeReview.lastReviewed;
+      }
+    }
+  }
+}
+
+export function getVUSData(vus: Vus[], excludeComments: boolean) {
+  const vusData = _.cloneDeep(vus);
+  const vusDataArray = [];
+  excludeComments = _.isBoolean(excludeComments) ? excludeComments : false;
+  for (const vusItem of vusData) {
+    if (excludeComments) {
+      delete vusItem.name_comments;
+    }
+    vusDataArray.push(vusItem);
+  }
+  return vusDataArray;
+}
+
+function shouldExclude(onlyReviewedContent: boolean, reviewObj: Review | undefined) {
```

Comment:
@calvinlu3 need your input here to check whether this still makes sense.

Reply:
@calvinlu3 ☝🏼

Reply:
Is there a use case for when `onlyReviewedContent` is false? If we're saving the data, we are currently skipping any data that still requires review.

I do believe that we still need a check to determine if `field_review` has a review object and the review object has any of the properties: `added, removed, demotedToVus, promotedToMutation, initialUpdate`

Reply:
@calvinlu3 we used to have a function to download a PDF which uses the latest data not reviewed content. That said, if we to implement the same functionality, I would download the review data instead. I can't think of any use case which onlyReviewedContent is false.

Reply:
That make sense then. We should probably keep this logic then in case we need to query for all (including items pending review).

However I would modify the condition to be
```
reviewObj && ((onlyReviewedContent && reviewObj.added === true && reviewObj.promotedToMutation === true && reviewObj.initialUpdate === true) ||
```
those are two new properties added to Review object that when true, also means that the content has not been reviewed yet and should be excluded